free-2.md

```diff
@@ -11,41 +11,100 @@ _What need is met by your idea? Who are you helping? What is that person's
+to one of simply thinking about the nature of the recurrence, expressing that,
+and letting the language transform it into an effective implementation.
+
+The most likely errors would be those from an ill formed recursive
+expression, for which we would rely on the host language's error handling).
 
 ### Why a language?
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+The key to solving an algorithm with dynamica programming is understanding
+the recurrence relation, and thus being able to build up a solution of a
+given size out of solutions of smaller sizes.
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+I am excited for this project to be able to generate code for computing
+solutions to generalized dynamic programming problems, especially
+getting it to work for some of the more difficult dynamic programming
+solutions to write by hand, such as the Nussinov Algorithm for RNA folding.
 
 ### Domain
 _Describe the project's domain in five words._
 
+Transform recurrences to dynamic programs.
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
-like? Why is this the right way to interact with the problem domain?_ 
+like? Why is this the right way to interact with the problem domain?_
+
+The user would likely use this language as an internal DSL of another language
+so that we can rely on the complexity and expressiveness of mathematical
+functions in the host language.  This will allow us to specify the general
+form of a recurrence and there will also be some manner of describing the
+order in which we will fill in our dynamic programming table (i.e. row-by-row
+versus along the diagonals).  This could also be described as a mathematical
+sequence or a function and an initial condition, as a method to get the
+order of tha table's cells to be filled in.
 
+This is a good way to approach the problem domain, since the problem is
+fundamentally one of computation, the idea being that we draw from the
+host language to make the recursive solution easier to describe.
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+I have a design decision to make, and I have not made it yet.  When the
+code is run, we will either:
+ * Generate the code which runs the DP and outputs it in a file.
+ * Generates and executes the DP immediately, returning an answer.
+I am currently leaning more toward the first option, as it would allow the
+dynamic programming algorithm to be reused rather than regenerated each time
+and also included in other code rather than run by hand.  Of course we could
+always offer both options, but implementing the first one has interesting
+design decisions related to pretty printing and comments in the generated code.
```

Comment:
I think if the first option was easily human readable that would be preferable, however I suspect using a computer to generate human readable code is rather difficult so perhaps the second option would make implementation easier and let you focus on design a bit more.